src/BuildToolchains.jl

```diff
@@ -253,3 +253,29 @@ function generate_toolchain_files!(platform::AbstractPlatform, envs::Dict{String
         end
     end
 end
+
+function cargo_config_file!(dir::AbstractString)
+    # Generate "${CARGO_HOME}/config.toml" file for Cargo where we give it the
+    # linkers for all our targets
+    open(joinpath(dir, "config.toml"), "w") do io
+        write(io, """
+        # Configuration file for `cargo`
+        """)
+        for platform in supported_platforms()
+            # Use `aatriplet` for the linker to match how the wrappers are
+            # written in
+            # https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/30d056ef68f81dca9cb91ededcce6b68c6466b37/src/Runner.jl#L599.
+            write(io, """
+            [target.$(map_rust_target(platform))]
+            linker = "$(aatriplet(platform))-gcc"
+            """)
+            if platforms_match(platform, Platform("aarch64", "linux"; libc="musl"))
+                # Work around bug for this platform:
+                # https://github.com/rust-lang/rust/issues/46651#issuecomment-433611633
+                write(io, """
+                rustflags = ["-C", "link-arg=-lgcc"]
+                """)
```

Comment:
Memo to self: check whether this is still needed after having upgraded Rust to 1.56.1